src/decorators/grpc-service.decorator.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback4-extension-starter
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+import {GRPCParams} from '../../';
+const grpc = require('grpc');
+//onst path = require('path');
+/**
+ * @function gRPCService
+ * @author Jonathan Casarrubias <t: johncasarrubias>
+ * @param params
+ * @license MIT
+ * @description This decorator provides a way to
+ * configure GRPC Micro Services within LoopBack 4 
+ * 
+ * Example of usage:
+ * 
+ * myproject/controllers/Greeter.ts
+ * 
+ * @gRPCService({
+ *   host: '127.0.0.1',
+ *   port: 3001,
+ *   proto: 'greeting.proto',
+ *   package: 'mypackage',
+ *   service: 'Greeter',
+ *   rpcs: ['sayHello']
+ * })
+ * class Greeter {
+ *   public sayHello(call, callback) {
+ *      callback(null, {message: 'Hello ' + call.request.name});
+ *   }
+ * } 
+ */
+export function gRPCService(params: GRPCParams) {
+  return function f(target: {new (): {}}) {
+    return <any>function ff() {
+      const proto = grpc.load(params.proto)[params.package];
+      const server = new grpc.Server();
+      const instance: {[key: string]: {}} = new target();
+      const handlers: {[key: string]: {}} = {};
+      params.rpcs.forEach((rpc: string) => (handlers[rpc] = instance[rpc]));
+      server.addService(proto[params.service].service, handlers);
+      server.bind(
+        `${params.host}:${params.port || 3000}`,
+        grpc.ServerCredentials.createInsecure(),
+      );
+      server.start();
```

Comment:
Uh oh, this approach is very different from the way how the rest of LoopBack Next has been designed so far. I'll need more time to grok through this and provide a meaningful advice on how to do this better.

Reply:
ok cool, I have some knowledge gaps, for instance supporting sequences.

Anyway if we decide to have just a single server, this decorator might work just to register the service, but not to start a new server....